src/site/content/en/fast/codelab-optimize-third-party-javascript/index.md

```diff
@@ -0,0 +1,248 @@
+
+The sample app has a YouTube video embedded in an [iframe](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe). To check out how many requests the page makes and which come from the embedded YouTube iframe:
+
+{% Instruction 'devtools-network', 'ol' %}
+{% Instruction 'disable-cache', 'ol' %}
+1. Select *Fast 3G* in the **Throttling** dropdown menu.
+{% Instruction 'reload-page', 'ol' %}
+
+![Screenshot of DevTools Network panel.](network.png)
+
+The **Network** panel reveals that the page made a total of 28 requests and loaded almost 1 MB of compressed resources.
+
+To identify the requests that the YouTube `iframe` made, look for the video ID `6lfaiXM6waw` in the **Initiator** column. To group together all the requests by domain:
+
+* In the **Network** panel, right-click a column title.
+
+* In the dropdown menu, select the **Domains** column.
+
+* To sort the requests by domain, click the **Domains** column title.
+
+The new sorting reveals that there are additional requests to Google domains. In total, the YouTube iframe makes 14 requests for scripts, stylesheets, images, and fonts. But unless users actually scroll down to play the video, they don’t really need all those assets.
+
+By waiting to lazy-load the video until a user scrolls down to that section of the page, you cut the number of requests the page initially makes. This approach saves users’ data and speeds up the initial load. 
+
+One way to implement lazy-loading is by using the [Intersection Observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API), a browser API that notifies you when an element enters or exits the browser's viewport. 
+
+### Step 1: Prevent video from loading initially
+
+To lazy-load the video iframe, you must first prevent it from loading in the usual way. Do that by replacing the `src` attribute with the `data-src` attribute to specify the video URL:
+
+```html
+<iframe width="560" height="315" data-src="https://www.youtube.com/embed/lS9D6w1GzGY" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+```
+
+`data-src` is a [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), which allows you to store extra information on standard HTML elements. A data attribute can be named anything, as long as it starts with "data-". 
+
+An iframe without a `src` simply won't load. 
+
+### Step 2: Use Intersection Observer to lazy-load the video
+
+To load the video when a user scrolls to it you need to know when that happens. That’s where the Intersection Observer API steps in. The Intersection Observer API lets you register a callback function that's executed whenever an element you want to track enters or exits the viewport.
+
+To get started, create a new file and name it `lazy-load.js`:
+
+{% Instruction 'create' %}
```

Comment:
I think it's okay because the UI won't let you _not name it_ and it's fairly obvious where the file name goes. 

We could also say "Click New File and give it a name." but do we really want to err on the side of overexplaining in this straightforward case?

Reply:
Yeah, sorry to be a pain, but I do think enumerated steps need to be fully accurate. I'm OK with the solution you propose so we can preserve this as a shortcode :)

Reply:
cc @robdodson just an FYI that I'll need to update Instruction.js